File: CustomShare/src/main/java/abhishekti/appyhigh/customshare/Builder.java
package abhishekti.appyhigh.customshare;


import abhishekti.appyhigh.customshare.models.Config;
import abhishekti.appyhigh.customshare.models.CustomShareData;

/**
 * Created by Abhishek Tiwari on 19-01-2021.
 */
public class Builder {

    private CustomShareData shareData;
    private CustomShare customShare;
    private Config config;

    public Builder(CustomShare customShare){
        this.customShare = customShare;
        this.config = new Config();
    }

    public Builder setShareData(CustomShareData data){
        this.shareData = data;
        return this;
    }

    public void show(){
        this.config.setCustomShareData(shareData);
        customShare.show(config);
    }
}

File: CustomShare/src/main/java/abhishekti/appyhigh/customshare/Builder.kt
package abhishekti.appyhigh.customshare

import abhishekti.appyhigh.customshare.models.Config
import abhishekti.appyhigh.customshare.models.CustomShareData

/**
 * Created by Abhishek Tiwari on 02-03-2021.
 */
class Builder(private var customShare: CustomShare) {

    private var customShareData: CustomShareData? = null
    private var config: Config = Config()
    private var isLocalFile: Boolean = false

    /**
     * Specifies whether the file is local to the app or not.
     * If this is set to true, then the library uses the FileProvider to share files since this is
     * the new standard for sharing internal files. An exception (URI Exposed exception)
     * is thrown if File Provider is not set
     * for sharing internal files/
     */
    fun isLocalFile(isLocal: Boolean): Builder{
        this.isLocalFile = isLocal
        return this
    }

    /**
     * Sets the data class for share intent
     */
    fun setShareData(data: CustomShareData): Builder{
        this.customShareData = data
        return this
    }

    fun show(){
        this.config.setCustomShareData(customShareData)
        this.config.setIsLocalFile(isLocalFile)
        customShare.show(config)
    }

}